### PR TITLE
sw_engine raster: code refactoring & optimize code.

### PR DESCRIPTION
**Pull request by hermet**
1. move the computation out of rolling if possible.
2. renamed internal variables & function prototypes.

- Description :

- Issue Tickets (if any) :

- Tests or Samples (if any) :

- Screen Shots (if any) :

### CODE REVIEW

**Comment by JSUYA** [comment]
HI
image

Please check demo screenshot

[Demo Code]
```
diff --git a/src/examples/PixelImage.cpp b/src/examples/PixelImage.cpp
index 61d1300..d710833 100644
--- a/src/examples/PixelImage.cpp
+++ b/src/examples/PixelImage.cpp
@@ -43,7 +43,7 @@ void tvgDrawCmds(tvg::Canvas* canvas)
     circle->appendCircle(350, 350, 200,200);
     circle->fill(255, 255, 255, 255);
 
-    picture2->composite(move(circle), tvg::CompositeMethod::ClipPath);
+    //picture2->composite(move(circle), tvg::CompositeMethod::ClipPath);
 
     canvas->push(move(picture2));
 }
```

File: src/lib/sw_engine/tvgSwRaster.cpp [hunk 1]
```diff
@@ -142,141 +142,104 @@ static bool _rasterTranslucentRle(SwSurface* surface, SwRleData* rle, uint32_t c
...
     }
     return true;
 }
 
 
-static bool _rasterTranslucentImage(SwSurface* surface, uint32_t *img, uint32_t opacity,const SwBBox& region, const Matrix* invTransform, uint32_t w, uint32_t h)
+static bool _rasterTranslucentImage(SwSurface* surface, uint32_t *img, uint32_t w, uint32_t h, uint32_t opacity, const SwBBox& region, const Matrix* invTransform)
 {
     for (auto y = region.min.y; y < region.max.y; y++) {
+        auto dst = &surface->buffer[y * surface->stride + region.min.x];
```
Comment (JSUYA) [inline src/lib/sw_engine/tvgSwRaster.cpp:193]: '+x' is missing from the original source.
Could you check this?

**Review by JSUYA — COMMENTED** [review]
Please check my comment

File: src/lib/sw_engine/tvgSwRaster.cpp [hunk 2]
```diff
@@ -142,141 +142,104 @@ static bool _rasterTranslucentRle(SwSurface* surface, SwRleData* rle, uint32_t c
...
     return true;
 }
 
 
-static bool _rasterImage(SwSurface* surface, uint32_t *img, const SwBBox& region, uint32_t w, uint32_t h)
+static bool _rasterImage(SwSurface* surface, uint32_t *img, uint32_t w, uint32_t h, const SwBBox& region)
 {
     for (auto y = region.min.y; y < region.max.y; y++) {
+        auto dst = &surface->buffer[y * surface->stride + region.min.x];
+        auto src = img + region.min.x + (y * w);    //TODO: need to use image's stride
```
Comment (JSUYA) [inline src/lib/sw_engine/tvgSwRaster.cpp:213]: x to region.min.x is it correct?
Please check that '+ x'  is also missing here.

File: src/lib/sw_engine/tvgSwRaster.cpp (hunk 1 above)
Comment (hermet) [reply src/lib/sw_engine/tvgSwRaster.cpp:191]: missed this case:
if (rX < 0 || rX >= w || rY < 0 || rY >= h) continue;

**Review by hermet — COMMENTED** [review]
x is handled by ++dst; not necesary.

File: src/lib/sw_engine/tvgSwRaster.cpp (hunk 2 above)
Comment (hermet) [reply src/lib/sw_engine/tvgSwRaster.cpp:210]: not necessary.

**Review by JSUYA — APPROVED** [review]
LGTM

**Pull request merged**